### PR TITLE
feat: azure kms provider as static pod

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Configures azure kms provider to be run as a static pod. The following change which was introduced prior to 1.13 ensures the api server waits for the kms plugin to be up and running - https://github.com/kubernetes/kubernetes/pull/68585

Things to complete as part of this PR:
- [x] Update tests
- [x] Update docs
- [x] Check cluster upgrade scenario

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [x] tested upgrade from previous version

**Notes**:
Tested cluster upgrade:
1. Created an aks-engine cluster with `enableEncryptionWithExternalKms: true`.
2. Validated secret in etcd is encrypted after cluster is deployed and secret is created.
3. Upgraded the cluster to a newer version using custom aks-engine build with the static pod changes.
4. Validated the secret is still encrypted after upgrade, the new kms provider static pod is running and `kubectl get secrets` returns the decoded content.

```bash
➜ kubectl get pods -n kube-system -l component=azure-kms-provider
NAME                                       READY   STATUS    RESTARTS   AGE
azure-kms-provider-k8s-master-14691719-0   1/1     Running   0          33m
```

### CODE REVIEW

**Comment by aramase** [comment]
This PR is ready for review. 

cc @ritazh

File: parts/k8s/cloud-init/masternodecustomdata.yml [hunk 1]
```diff
@@ -159,13 +159,6 @@ write_files:
   content: !!binary |
     {{CloudInitData "labelNodesSystemdService"}}
 
-- path: /etc/systemd/system/kms.service
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/masternodecustomdata.yml:162]: So basically we are deprecating all systemd foo in favor of everything being deliverd via daemonset?

Comment (jackfrancis) [reply parts/k8s/cloud-init/masternodecustomdata.yml:162]: Actually nevermind I see it's a system Pod

File: pkg/engine/const.go [hunk 2]
```diff
@@ -113,7 +113,6 @@ const (
 	kubernetesMasterGenerateProxyCertsScript = "k8s/cloud-init/artifacts/generateproxycerts.sh"
 	kubernetesCustomSearchDomainsScript      = "k8s/cloud-init/artifacts/setup-custom-search-domains.sh"
 	kubeletSystemdService                    = "k8s/cloud-init/artifacts/kubelet.service"
-	kmsSystemdService                        = "k8s/cloud-init/artifacts/kms.service"
```
Comment (jackfrancis) [inline pkg/engine/const.go:116]: We should be able to delete the `parts/k8s/cloud-init/artifacts/kms.service` file entirely from the codebase now

Comment (aramase) [reply pkg/engine/const.go:116]: deleted `parts/k8s/cloud-init/artifacts/kms.service`

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**